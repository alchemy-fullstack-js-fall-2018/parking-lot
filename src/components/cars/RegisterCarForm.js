import React, { Fragment, Component } from 'react';
import PropTypes from 'prop-types';

export default class RegisterCarForm extends Component {
  static propTypes = {
    registerCar: PropTypes.func
  };

  state = {
    id: '',
    state: '',
    make: '',
    model: '',
    color: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = event => {
    const { id, state, make, model, color } = this.state;
    console.log('event', event);
    event.preventDefault();
    this.props.registerCar({ id, state, make, model, color });
  };

  render() {
    const { id, state, make, model, color } = this.state;

    return (
      <Fragment>
        <h2>Registration Form</h2>
        <form onSubmit={this.handleSubmit}>
          <label>
            Licence Plate:
            <input value={id} name="id" onChange={this.handleChange} />
          </label>
          <label>
            Registration State:
            <input value={state} name="state" onChange={this.handleChange} />
          </label>
          <label>
            Make:
            <input value={make} name="make" onChange={this.handleChange} />
          </label>
          <label>
            Model:
            <input value={model} name="model" onChange={this.handleChange} />
          </label>
          <label>
            Color:
            <input value={color} name="color" onChange={this.handleChange} />
          </label>
          <button>Submit</button>
        </form>
      </Fragment>
    );
  }
}
