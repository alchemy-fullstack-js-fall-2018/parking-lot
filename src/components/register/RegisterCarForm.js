import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';
import { ROUTES } from '../../routes';

export default class RegisterCarForm extends PureComponent {
  static propTypes = {
    registerCar: PropTypes.func
  };

  state = {
    plate: '',
    make: '',
    model: ''
  };

  handleSubmit = e => {
    e.preventDefault();

    const { plate, make, model } = this.state;
    const { registerCar } = this.props;

    registerCar(plate, make, model);
    this.props.history.push(ROUTES.CAR_DETAIL.linkTo(plate));
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value.toUpperCase() });
  };

  render() {
    return (
      <Fragment>
        <h3>Add Your Junker to the Lot!</h3>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor="plate">Plate Info</label>
          <input
            id="plate"
            type="text"
            name="plate"
            onChange={this.handleChange}
          />

          <label htmlFor="make">Make</label>
          <input
            id="make"
            type="text"
            name="make"
            onChange={this.handleChange}
          />

          <label htmlFor="model">Model</label>
          <input
            id="model"
            type="text"
            name="model"
            onChange={this.handleChange}
          />

          <button>Good Luck!</button>
        </form>
      </Fragment>
    );
  }
}
