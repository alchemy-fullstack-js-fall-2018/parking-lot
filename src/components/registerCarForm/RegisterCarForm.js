import React, { PureComponent, Fragment } from 'react';
import PropTypes from 'prop-types';

export default class RegisterCarForm extends PureComponent {

  static propTypes = {
    registerCar: PropTypes.func
  };

  state = {
    make: '',
    model: '',
    plate: ''
  };

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  };

  handleSubmit = event => {
    const { registerCar } = this.props;
    const { make, model, plate } = this.state;

    event.preventDefault();
    registerCar(make, model, plate);
    this.props.history.push(`/cars/${plate}`);
  };

  render() {

    return (
      <Fragment>
        <h1>Register a New Car</h1>
        <form onSubmit={this.handleSubmit}>
          <label htmlFor='make'>Make</label>
          <input id='make' type='text' name='make' onChange={this.handleChange}/>

          <label htmlFor='model'>Model</label>
          <input id='model' type='text' name='model' onChange={this.handleChange}/>

          <label htmlFor='plate'>Plate Info</label>
          <input id='plate' type='text' name='plate' onChange={this.handleChange}/>

          <button>Register</button>
        </form>
      </Fragment>
    );
  }
}
