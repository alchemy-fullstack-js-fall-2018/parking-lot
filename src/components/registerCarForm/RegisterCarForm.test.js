import React from 'react';
import { shallow } from 'enzyme';
import RegisterCarForm from './RegisterCarForm';
jest.mock('../../routes/routes');

describe('RegisterCarForm component', () => {
  it('renders a component', () => {
    const wrapper = shallow(<RegisterCarForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it('submits a cars make, model, and plate onSubmit', () => {
    const registerCar = jest.fn();
    const wrapper = shallow(
      <RegisterCarForm history={[]} registerCar={registerCar} />
    );

    wrapper.find('#make').simulate('change', { target: { name: 'make', value: 'Toyota' } });
    wrapper.find('#model').simulate('change', { target: { name: 'model', value: 'Corolla' } });
    wrapper.find('#plate').simulate('change', { target: { name: 'plate', value: 'ABC123' } });
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    expect(registerCar.mock.calls[0]).toEqual(['Toyota', 'Corolla', 'ABC123']);
  });
});
