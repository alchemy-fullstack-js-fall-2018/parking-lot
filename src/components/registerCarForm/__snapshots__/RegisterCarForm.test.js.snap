// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterCarForm component renders a component 1`] = `
<Fragment>
  <h1>
    Register a New Car
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="make"
    >
      Make
    </label>
    <input
      id="make"
      name="make"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="model"
    >
      Model
    </label>
    <input
      id="model"
      name="model"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="plate"
    >
      Plate Info
    </label>
    <input
      id="plate"
      name="plate"
      onChange={[Function]}
      type="text"
    />
    <button>
      Register
    </button>
  </form>
</Fragment>
`;
